src/device/arm/arm.go

```diff
@@ -174,20 +175,13 @@ func SetPriority(irq uint32, priority uint32) {
 
 // DisableInterrupts disables all interrupts, and returns the old interrupt
 // state.
-func DisableInterrupts() uintptr {
-	return AsmFull(`
-		mrs {}, PRIMASK
-		cpsid i
-	`, nil)
-}
+//go:linkname DisableInterrupts DisableInterrupts
+func DisableInterrupts() uintptr
```

Comment:
Please use `//export` instead of `//go:linkname`. The reason is that `//go:linkname` uses the TinyGo calling convention, while `//export` uses the C calling convention. There might be a small difference between them.